tests/Browser/LoginTest.php

```diff
@@ -0,0 +1,234 @@
+<?php
+
+namespace Tests\Browser;
+
+use Illuminate\Foundation\Testing\DatabaseMigrations;
+use Laravel\Dusk\Browser;
+use Tests\DuskTestCase;
+use App\User;
+use Hash;
+
+class LoginTest extends DuskTestCase
+{
+    /**
+    * A basic browser url not login.
+    */
+    public function testPathWrong()
+    {
+        $this->browse(function (Browser $browser) {
+            $browser->visit('/abc')
+                    ->assertSee('404');
+        });
+    }
+
+    /**
+    * A basic browser go to path admin dislay Login.
+    */
+    public function testPathAdmin()
+    {
+        $this->browse(function (Browser $browser) {
+            $browser->visit('/admin')
+                    ->assertPathIs('/login');
```

Comment:
Look like '/admin' and '/login' url call multiple times.
Do we need to create a Constants class that contains these url values?

Reply:
i refacted code.

Reply:
As i comment meaning:
Should we need to create Constants class and store '/admin' and '/login' and use that constants in every place need it?
Ex: We can use '/admin' for:
+ This test file
+ All blade html file have href with link go to '/admin'
+ Others need...
--------------------------------------------------------------
In your fix it will:
+ Only local constant of test/Browser/LoginTest.php and can use in other place.
+ Unreasonable content structure(Constant content use public but store in 'test/' folder).
--------------------------------------------------------------------------------------------------------
And last, this is my comment question, not requirement so you can review and evaluation it.
+ If it useful, you should refacted it.
+ If it unnecessary, only need give me a comment explain why.

Reply:
I was created file config/contants.php to faceted